ui/src/BsddTreeView.js

```diff
@@ -0,0 +1,147 @@
+import Table from '@mui/material/Table';
+import TableBody from '@mui/material/TableBody';
+import TableCell from '@mui/material/TableCell';
+import TableContainer from '@mui/material/TableContainer';
+import TableHead from '@mui/material/TableHead';
+import TableRow from '@mui/material/TableRow';
+import Paper from '@mui/material/Paper';
+import { statusToColor } from './mappings'
+
+export default function BsddTreeView({ bsddResults, status }) {
+
+  const bsdd = bsddResults;
+
+  return (
+    <Paper sx={{overflow: 'hidden'}}><TreeView
+      aria-label="file system navigator"
+      defaultCollapseIcon={<ExpandMoreIcon />}
+      defaultExpandIcon={<ChevronRightIcon />}
+      defaultExpanded={["0"]}
+      sx={{ "width": "850px", "backgroundColor": statusToColor[status], "> li > .MuiTreeItem-content": { padding: "16px" }, ".MuiTreeItem-content.Mui-expanded": { borderBottom: 'solid 1px black' } }}
+    >
+      <TreeItem nodeId="0" label={"bSDD"}>
+        <TreeView defaultCollapseIcon={<ExpandMoreIcon />}
+          defaultExpandIcon={<ChevronRightIcon />}>
+          {
+            Object.keys(bsdd["bsdd"]).map((domain, d_index) => {
+              return <TreeItem nodeId={d_index} label={`Domain: ${domain}`}>
+                <TreeView defaultCollapseIcon={<ExpandMoreIcon />}
+                  defaultExpandIcon={<ChevronRightIcon />}>
+                  {
+                    Object.keys(bsdd["bsdd"][domain]).map((classification, c_index) => {
+                      return <TreeItem nodeId={c_index} label={`Classification: ${classification}`}>
+                        {
+                          Object.keys(bsdd["bsdd"][domain][classification]).map((result) => {
+                            return <div >
+                              <br></br>
+                              <TableContainer sx={{
+                                minWidth: 650,
+                                "width": "90%",
+                                "padding": "10px"
+                              }} >
+                                <Table sx={{
+                                  minWidth: 650,
+                                  "backgroundColor": "rgb(238, 238, 238)",
+                                }}
+                                  size="small"
+                                  aria-label="a dense table">
+                                  <TableHead>
+                                    <TableRow>
+                                      <TableCell align="center">Instance</TableCell>
+                                      <TableCell align="center">Requirement</TableCell>
+                                      <TableCell align="center">Required</TableCell>
+                                      <TableCell align="center">Observed</TableCell>
+
+                                    </TableRow>
+                                  </TableHead>
+                                  <TableBody>
+
+                                    {/* IFC TYPE */}
```

Comment:
Let's keep it like this for now, but repetition here gives me the creeps a bit. I'd rather have one `TableRow` in the code in a for loop.

Reply:
improved in part in https://github.com/AECgeeks/ifc-pipeline-validation/pull/51/commits/06cafc8f2df3e6c7a678e5e21642213f80f8793d